src/eva/vision/data/datasets/segmentation/lits.py

```diff
@@ -0,0 +1,178 @@
+"""LiTS dataset."""
+
+import functools
+import glob
+import os
+from typing import Any, Callable, Dict, List, Literal, Tuple
+
+import torch
+from torchvision import tv_tensors
+from typing_extensions import override
+
+from eva.core import utils
+from eva.vision.data.datasets import _utils as data_utils
+from eva.vision.data.datasets import _validators
+from eva.vision.data.datasets.segmentation import base
+from eva.vision.utils import io
+
+
+class LiTS(base.ImageSegmentation):
+    """LiTS - Liver Tumor Segmentation Challenge.
+
+    Webpage: https://competitions.codalab.org/competitions/17094
+
+    For the splits we follow: https://arxiv.org/pdf/2010.01663v2
+    """
+
+    _train_index_ranges: List[Tuple[int, int]] = [(0, 102)]
+    _val_index_ranges: List[Tuple[int, int]] = [(102, 117)]
+    _test_index_ranges: List[Tuple[int, int]] = [(117, 131)]
+    """Index ranges per split."""
+
+    _sample_every_n_slices: int | None = None
+    """The amount of slices to sub-sample per 3D CT scan image."""
+
+    _expected_dataset_lengths: Dict[str | None, int] = {
+        "train": 39307,
+        "val": 12045,
+        "test": 7286,
+        None: 58638,
+    }
+    """Dataset version and split to the expected size."""
+
+    _license: str = (
+        "Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License "
+        "(https://creativecommons.org/licenses/by-nc-nd/4.0/deed.en)"
+    )
+    """Dataset license."""
+
+    def __init__(
+        self,
+        root: str,
+        split: Literal["train", "val", "test"] | None = None,
+        transforms: Callable | None = None,
+    ) -> None:
+        """Initialize dataset.
+
+        Args:
+            root: Path to the root directory of the dataset. The dataset will
+                be downloaded and extracted here, if it does not already exist.
+            split: Dataset split to use.
+            transforms: A function/transforms that takes in an image and a target
+                mask and returns the transformed versions of both.
+        """
+        super().__init__(transforms=transforms)
+
+        self._root = root
+        self._split = split
+
+        self._indices: List[Tuple[int, int]] = []
+
+    @property
+    @override
+    def classes(self) -> List[str]:
+        return ["liver", "tumor"]
```

Comment:
In `consep.py` we list "background" as a class here as well. If not doing so, I think the mapping you get in `class_to_idx` will be wrong (the idx there should refer to the values in the target masks).

Reply:
this is something that we have to sign, as in the other datasets the background is not one of the classes - its getting a bit weirded when the mask is not semantic labels (HxW) but a concat of binary one-hot (NUM_CLASSESxHxW), then the first channel represents the class 😅 . background is the absents of class so its not defined as a class if this makes sense, its always the 0 and datasets dont really refer to it

but we do want a consisted behaviour here, so probably I would say to include the background in all, but let me see how other are handling this

Reply:
I added as background for now, I'll open an issue to discuss more about it